packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.styles.ts

```diff
@@ -111,25 +112,68 @@ export const getArrowButtonStyles = memoizeFunction((
   return concatStyleSets(
     defaultArrowButtonStyles,
     isUpArrow ? defaultUpArrowButtonStyles : defaultDownArrowButtonStyles,
-    customSpecificArrowStyles
+    customStyles
   ) as IButtonStyles;
 });
 
+export const getUpArrowButtonStyles = memoizeFunction((
+  props: ISpinButtonArrowStyleProps,
+): IButtonStyles => {
+  return getArrowButtonStyles(props, true);
+});
+
+export const getDownArrowButtonStyles = memoizeFunction((
+  props: ISpinButtonArrowStyleProps,
+): IButtonStyles => {
+  return getArrowButtonStyles(props, false);
+});
+
+const _getStyleForLabelBasedOnPosition = (
+  styles: ISpinButtonStyles, labelPosition?: Position
+): IStyle => {
+  switch (labelPosition) {
+    case Position.start:
+      return styles.labelWrapperStart;
+    case Position.end:
+      return styles.labelWrapperEnd;
+    case Position.top:
+      return styles.labelWrapperTop;
+    case Position.bottom:
+      return styles.labelWrapperBottom;
+    default: return {};
+  }
+};
+
+const _getStyleForRootBasedOnPosition = (styles: ISpinButtonStyles, labelPosition?: Position): IStyle => {
+  switch (labelPosition) {
+    case Position.top:
+    case Position.bottom:
+      return styles.spinButtonWrapperTopBottom;
+    default: return {};
+  }
+};
+
 export const getStyles = memoizeFunction((
-  theme: ITheme,
-  customStyles?: Partial<ISpinButtonStyles>
-): ISpinButtonStyles => {
+  props: ISpinButtonStyleProps
+): Partial<ISpinButtonStyles> => {

```

Comment:
This needs to return a Partial<ISpinButtonStyles>. With the move to getStyles, the list of styles that gets returned is concatenated. For example, Root:{} now contains everything that was in RootHovered, RootFocused, etc. I would just remove the unnecessary styles from ISpinButtonStyles, but this would break current users that are relying on the current model for passing in styles, which uses these other styles